### PR TITLE
Render different material for grinder tailings.

### PR DESCRIPTION
**Pull request by mogumbo**
To test:
* roslaunch ow atacama_y1a.launch
* rosrun ow_lander task_grind.py -x 1.75 -y 0 -z -0.25
* rosrun ow_lander pan_tilt_move_joints.py 1.2 0.8
* rosrun ow_lander camera_capture.py

The resulting image looks decent, but I'm not a big fan of the look of the new material. If you zoom in closely in gzclient it looks a lot more like shiny stucco than snow. I'll see if I can do a separate ticket to make things look better, maybe by adding some simple subsurface scattering technique.

### CODE REVIEW

**Comment by AstroStucky** [comment]
Test passes, and I am by no means the best person to assess this, but it looks great to me! It even has the glitter glints you get with snow.
image

File: models/atacama_y1a/materials/scripts/ow_terrain.frag [hunk 1]
```diff
@@ -206,12 +208,12 @@ void lighting(vec3 wsDirToSun, vec3 wsDirToEye, vec3 wsNormal, vec4 wsDetailNorm
...
   // directional light diffuse
   float sundiffuse = max(dot(wsDetailNormalHeight.xyz, wsDirToSun), 0.0);
-  diffuse += sunIntensity * (sundiffuse * visibility);
+  diffuse = sunIntensity * (sundiffuse * visibility);
 
   // directional light specular
   vec3 reflectvec = reflect(-wsDirToEye, wsDetailNormalHeight.xyz);
   float sunspec = pow(max(dot(wsDirToSun, reflectvec), 0.0), specular_power);
-  specular += sunIntensity * (sunspec * visibility);
+  specular = sunIntensity * (sunspec * visibility);
```
Comment (AstroStucky) [inline models/atacama_y1a/materials/scripts/ow_terrain.frag]: Was this just a semi-unrelated bug fix?

File: models/atacama_y1a/materials/scripts/ow_terrain.frag [hunk 2]
```diff
@@ -242,6 +244,52 @@ void lighting(vec3 wsDirToSun, vec3 wsDirToEye, vec3 wsNormal, vec4 wsDetailNorm
...
+vec3 triPlanarBlending(vec3 normal) 
+{
+  vec3 blend = normal * normal;
+  blend *= blend;
+  return blend / dot(blend, vec3(1,1,1));
+}
+
+/**
+ * Tri-planar texturing using reoriented normal mapping method from
+ * https://bgolus.medium.com/normal-mapping-for-a-triplanar-shader-10bf39dca05a
```
Comment (AstroStucky) [inline models/atacama_y1a/materials/scripts/ow_terrain.frag:259]: Unfortunate this is on Medium, I heard they're putting their content behind an account and maybe even a paywall soon :(

File: models/atacama_y1a/materials/scripts/ow_terrain.frag [hunk 3]
```diff
@@ -242,6 +244,52 @@ void lighting(vec3 wsDirToSun, vec3 wsDirToEye, vec3 wsNormal, vec4 wsDetailNorm
...
+ */
+vec3 triPlanarNormalMapping(vec3 wsPos, vec3 osNormal)
+{
+  vec2 xuv = wsPos.zy * 5.0;
+  vec2 yuv = wsPos.xz * 5.0;
+  vec2 zuv = wsPos.xy * 5.0;
```
Comment (AstroStucky) [inline models/atacama_y1a/materials/scripts/ow_terrain.frag:266]: Comment what this does, please.

File: models/atacama_y1a/materials/scripts/ow_terrain.frag [hunk 4]
```diff
@@ -242,6 +244,52 @@ void lighting(vec3 wsDirToSun, vec3 wsDirToEye, vec3 wsNormal, vec4 wsDetailNorm
...
+  // Invert Y coordinate to have correct addressing of texels. Otherwise, lighting is inverted.
+  xuv.y *= -1.0;
+  yuv.y *= -1.0;
+  zuv.y *= -1.0;
+
+  vec3 tsNormalX = texture2D(snowNormalMap, xuv).rgb * vec3(2.0) - vec3(1.0);
+  vec3 tsNormalY = texture2D(snowNormalMap, yuv).rgb * vec3(2.0) - vec3(1.0);
+  vec3 tsNormalZ = texture2D(snowNormalMap, zuv).rgb * vec3(2.0) - vec3(1.0);
```
Comment (AstroStucky) [inline models/atacama_y1a/materials/scripts/ow_terrain.frag:275]: This is also a mystery to me. Could it get a comment?

**Review by AstroStucky — APPROVED** [review]
I lack the expertise to fully understand the shader code, so I'm not terribly confident in my review of it. There were some operations I'd like to see commented on.

Even if the snow texture is not quite where you want it to be, this adds a lot of value to the visual simulation on it's own. Great contribution!

File: models/atacama_y1a/materials/scripts/ow_terrain.frag (hunk 1 above)
Comment (mogumbo) [reply models/atacama_y1a/materials/scripts/ow_terrain.frag]: Good eye. This doesn't actually change the output of the shader, but I reverted it back to += because the code is a little more robust that way.

File: models/atacama_y1a/materials/scripts/ow_terrain.frag (hunk 3 above)
Comment (mogumbo) [reply models/atacama_y1a/materials/scripts/ow_terrain.frag:266]: Comment added.

File: models/atacama_y1a/materials/scripts/ow_terrain.frag (hunk 4 above)
Comment (mogumbo) [reply models/atacama_y1a/materials/scripts/ow_terrain.frag:275]: Comment added.

**Review by AstroStucky — APPROVED** [review]
